### DISCUSSION TITLE
remote.txt changes not working

### DISCUSSION THREAD

**Discussion by gbjbasdw** [GerryDazoo/Slinger#425] [Q&A]
I don't know what is going on here.
I have made some changes to the remote.txt file e.g. I have changed the text of the "OK" button to "nOK" saved it and when I go to the remote in my browser (http://192.168.0.10:64000/remote/ProHD) it has not changed.

The config.ini file has "include=remote.txt" in the [REMOTE] section I have also tried include=c:\sliger\remote.txt in case it needed the full path but again no change.  Also the ProHD section has VideoSource=1 for composite input so the "code" parameter ties up with this.
[REMOTE]
;see release notes for changing this if your remote doesn't work, But as a start make it 
;the same as your configured VideoSource 0-3. See above.
code=1
;path to an external remote control definitions file, if the defaults
;aren't working for you or you want to change the format of the web page.
;use the supplied remote.txt as a starting point.
include=remote.txt

Here is the Sling Server log
Version : 3.08f Running on Windows-10-10.0.19044-SP0 pid= 7444 C:\Slinger\slingbox_server.exe
Using config file config.ini
Connection Manager Running on port 64000 with 10 max streams using URL sserver.
BOXES [('sb1', 'ProHD')]
Streamer Running:  10 config.ini ProHD ProHD 64000 8388608
starting up on  port 64000 8388608
Checking for slingbox at ('192.168.0.21', 5001)
('192.168.0.21', 5001) OK
ProHD Using slingbox at  ('192.168.0.21', 5001)
Streamer:  ProHD Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Building page for ProHD
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
12/07/2022, 10:47:50.186   RemoteControl connection from ('192.168.0.10', 59654)

Starting remote control stream handler for  ('192.168.0.10', 59654) to port 64001
Remote Control Connected
GET remote/ProHD
Exiting Remote Control Stream Handler for ('192.168.0.10', 59654)
12/07/2022, 10:47:50.573   RemoteControl connection from ('192.168.0.10', 59655)

Starting remote control stream handler for  ('192.168.0.10', 59655) to port 64001
Remote Control Connected
GET favicon.ico
Exiting Remote Control Stream Handler for ('192.168.0.10', 59655)  

I don't know why it starts and stops the remote control stream handler or if this is normal behaviour.

The remote displays in the browser ok but not with any of my changes showing.
I know there are a couple of fancy remotes for the TIVO but I was trying to get the simpler one working first because I could not see a "My shows" button used to display what I have recorded.
Any suggestions ?

**Comment by gbjbasdw** [comment]
I have just had a go with the new "custom remotes" way of using remotes.
Copied the html and directory to my "slinger" directory and changed the "include" parameter in the config file to =TIVORemotePF.html

Ran the slinger and got a load of new stuff for the keycodes See below
Version : 3.08f Running on Windows-10-10.0.19044-SP0 pid= 5880 C:\Slinger\slingbox_server.exe
Using config file config.ini
Connection Manager Running on port 64000 with 10 max streams using URL sserver.
BOXES [('sb1', 'ProHD')]
Streamer Running:  10 config.ini ProHD ProHD 64000 8388608
starting up on  port 64000 8388608
Checking for slingbox at ('192.168.0.21', 5001)
('192.168.0.21', 5001) OK
ProHD Using slingbox at  ('192.168.0.21', 5001)
Streamer:  ProHD Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Building page for ProHD
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
12/07/2022, 11:35:11.4   Streaming request from ('192.168.0.10', 60039)

Slinginfo  "Solo/Pro/ProHD" 16 30 192.168.0.21 5001 3000 1
ProHD Starting Stream for  192.168.0.10:60039
Connecting... ('192.168.0.21', 5001) Control
ProHD Selecting Video Source 1
Keycodes= [1, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 46, 47, 50, 74, 75, 76, 77]
VideoParameters: Resolution= 16 FrameRate= 30 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 32
Connecting... ('192.168.0.21', 5001) Stream
ProHD Stream started at 12/07/2022, 11:35:17.82  732 0
.12/07/2022, 11:35:28.198  ProHD Sending Start Channel 1
sending key 9 1
.12/07/2022, 11:35:41.385   RemoteControl connection from ('192.168.0.10', 60059)

Starting remote control stream handler for  ('192.168.0.10', 60059) to port 64001
Remote Control Connected
GET remote/ProHD
Exiting Remote Control Stream Handler for ('192.168.0.10', 60059)
......12/07/2022, 11:36:50.319   RemoteControl connection from ('192.168.0.10', 60067)

Starting remote control stream handler for  ('192.168.0.10', 60067) to port 64001
Remote Control Connected
GET favicon.ico
Exiting Remote Control Stream Handler for ('192.168.0.10', 60067)
.12/07/2022, 11:36:50.700  ProHD 1 Clients.192.168.0.10:60039
...12/07/2022, 11:37:28.329  ProHD Stream Terminated for  192.168.0.10:60039
ProHD Shutting down connections
ProHD Logging Out
Streamer:  ProHD Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox

Video stream is OK  in VLC

Opened Edge browser and typed
http://192.168.0.10:64000/remote/ProHD
QUESTION - Is this how I am supposed to be accessing the new remote control ? If not what am I supposed to use ?
What I got displayed was the old style remote but without any of my changes again. I just can't see where the hell it is picking this up from !

Can one of the coders have a look and point me in the right direction.

**Comment by gbjbasdw** [comment]
Sorry I forgot to say - what is the GET favicon.ico doing in the slinger log ? Is this what is getting displayed in the browser  perhaps ?

**Comment by GerryDazoo** [comment]
There is only one coder, me.

"Streamer: ProHD Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox"
You need to be streaming from the slingbox for the remote to work. And your logs show the stream has been shutdown.

All browsers ask for a favicon when it connects. If you had one it would display it on your remote tab in your browser. It can be safely ignored.

You can remove the code=1 from your remote section it's redundant if you've specified VideoSource.

I don't think you're actually using the config file you think you are. If you were you should see this log
Reading Custom Remote definition from TIVORemotePF.html

Please post your complete conifg,ini
